services/gateway_manager/src/main/java/com/futurewei/alcor/gatewaymanager/controller/GatewayController.java

```diff
@@ -0,0 +1,149 @@
+        RestPreconditionsUtil.verifyParameterNotNullorEmpty(projectId);
+        RestPreconditionsUtil.verifyParameterNotNullorEmpty(vpcId);
+        VerifyParameterUtils.checkGatewayInfo(gatewayInfoJson.getGatewayInfo());
+        gatewayService.updateGatewayInfoForZeta(projectId, gatewayInfoJson.getGatewayInfo());
+        log.info("GatewayInfo updated success,GatewayInfo is: {}", gatewayInfoJson.getGatewayInfo());
+        return new ResponseId(vpcId);
+    }
+
+    /**
+     * Delete a GatewayInfo (for zeta gateway)
+     *
+     * @param projectId
+     * @param vpcId
+     * @return
+     * @throws Exception
+     */
+    @DeleteMapping("/project/{projectid}/gatewayinfo/{resource_id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public ResponseId deleteGatewayInfoForZeta(@PathVariable("projectid") String projectId, @PathVariable("resource_id") String vpcId) throws Exception {
+        RestPreconditionsUtil.verifyParameterNotNullorEmpty(projectId);
+        RestPreconditionsUtil.verifyParameterNotNullorEmpty(vpcId);
+        gatewayService.deleteGatewayInfoForZeta(projectId, vpcId);
+        log.info("GatewayInfo deleted success,the resource_id is: {}", vpcId);
+        return new ResponseId(vpcId);
+    }
+
+
+    /**
+     * Create a gateway
+     *
+     * @param projectId
+     */
+    @PostMapping("/project/{projectid}/gateways")
+    @ResponseStatus(HttpStatus.CREATED)
+    public void createGateway(@PathVariable("projectid") String projectId) {
+
+    }
+
+    /**
+     * Update a gateway
+     *
+     * @param projectId
+     */
+    @PutMapping("/project/{projectid}/gateways")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void updateGateway(@PathVariable("projectid") String projectId) {
+
+    }
+
+    /**
+     * Update a gateway by ID
+     *
+     * @param projectId
+     * @param gatewayId
+     * @return
+     */
+    @PutMapping("/project/{projectid}/gateways/{gateway_id}")
+    public void updateGatewayById(@PathVariable("projectid") String projectId, @PathVariable("gateway_id") String gatewayId) {
+
+    }
+
+    /**
+     * Query a gateway’s state
+     *
+     * @param projectId
+     * @param gatewayId
+     * @return
+     */
+    @GetMapping("/project/{projectid}/gateways/{gateway_id}")
+    public GatewayEntity getGatewayStateById(@PathVariable("projectid") String projectId, @PathVariable("gateway_id") String gatewayId) throws Exception {
+        RestPreconditionsUtil.verifyParameterNotNullorEmpty(projectId);
+        RestPreconditionsUtil.verifyParameterNotNullorEmpty(gatewayId);
+        return gatewayService.getGatewayEntityById(gatewayId);
+    }
+
+    /**
+     * List All Available Gateways
+     *
+     * @param projectId
+     * @return
+     */
+    @GetMapping("/project/{projectid}/gateways")
+    public void getGatewaysByVpcId(@PathVariable("projectid") String projectId) {
+
```

Comment:
@songxiaoyan The gateway here should be project-scope, not vpc-scope. So, it should be listed all available gateways in the current project.

Reply:
This is a reserved interface, which will be adjusted according to actual business in the future